`Index(MutableSequence[tuple])` should give `MultiIndex`

## Describe the bug
A clear and concise description of what the bug is.

## To Reproduce

### Provide a minimal runnable `pandas` example

```py
from typing import reveal_type
import pandas as pd
from collections import UserList, deque

data = [(1,), (2,)]

# runtime, all is MultiIndex
reveal_type(pd.Index(data))  # static: Index
reveal_type(pd.Index(UserList(data)))  # static: Index
reveal_type(pd.Index(deque(data)))  # static: not supported
```
### Indicate which type checker you are using

both

### Show the error message received from that type checker while checking your example.

#### `mypy`

```text
ttest.py:8: note: Revealed type is "pandas.core.indexes.base.Index[Any]"
ttest.py:9: note: Revealed type is "pandas.core.indexes.base.Index[Any]"
ttest.py:10: note: Revealed type is "pandas.core.indexes.base.Index[builtins.bool]"
ttest.py:10: error: Argument 1 to "deque" has incompatible type "list[tuple[int]]"; expected "Iterable[builtins.bool | numpy.bool[builtins.bool]]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

#### `pyright`

```text
ttest.py
  ttest.py:8:13 - information: Type of "pd.Index(data)" is "Index[Any]"
  ttest.py:9:13 - information: Type of "pd.Index(UserList(data))" is "Index[Any]"
  ttest.py:10:13 - error: No overloads for "__new__" match the provided arguments (reportCallIssue)
  ttest.py:10:13 - information: Type of "pd.Index(deque(data))" is "Index[Any]"
  ttest.py:10:22 - error: Argument of type "deque[tuple[int]]" cannot be assigned to parameter "data" of type "AxesData[Unknown]" in function "__new__"
    Type "deque[tuple[int]]" is not assignable to type "AxesData[Unknown]"
      "deque[tuple[int]]" is not assignable to "Mapping[Unknown, Any]"
      "deque[tuple[int]]" is not assignable to "ExtensionArray"
      "deque[tuple[int]]" is not assignable to "ndarray[_AnyShape, dtype[Any]]"
      "deque[tuple[int]]" is not assignable to "Index[Any]"
      "deque[tuple[int]]" is not assignable to "Series[Any]"
      "deque[tuple[int]]" is incompatible with protocol "SequenceNotStr[Any]"
        "__getitem__" is an incompatible type
    ... (reportArgumentType)
  ttest.py:10:22 - error: Argument of type "deque[tuple[int]]" cannot be assigned to parameter "data" of type "AxesData[Unknown]" in function "__new__"
    Type "deque[tuple[int]]" is not assignable to type "AxesData[Unknown]"
      "deque[tuple[int]]" is not assignable to "Mapping[Unknown, Any]"
      "deque[tuple[int]]" is not assignable to "ExtensionArray"
      "deque[tuple[int]]" is not assignable to "ndarray[_AnyShape, dtype[Any]]"
      "deque[tuple[int]]" is not assignable to "Index[Any]"
      "deque[tuple[int]]" is not assignable to "Series[Any]"
      "deque[tuple[int]]" is incompatible with protocol "SequenceNotStr[Any]"
        "__getitem__" is an incompatible type (reportArgumentType)
3 errors, 0 warnings, 3 informations
```

## Please complete the following information:

 - OS: Windows
 - OS Version: 11
 - python version: 3.14
 - version of type checker: mypy 1.18.2, pyright 1.1.407
 - version of installed `pandas-stubs`: 5a47589c9b65f473229799b660b6d8fa6cc5a155


## Additional context

We need to fix `Index.__new__`.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -72,7 +72,7 @@ jobs:
 
   type_completeness:
     runs-on: ubuntu-latest
-    timeout-minutes: 5
+    timeout-minutes: 10
     permissions:
       contents: read
       actions: write

--- a/pandas-stubs/core/indexes/base.pyi
+++ b/pandas-stubs/core/indexes/base.pyi
@@ -318,7 +318,7 @@ class Index(IndexOpsMixin[S1], ElementOpsMixin[S1]):
     @overload
     def __new__(
         cls,
-        data: Sequence[tuple[Any, ...]],
+        data: Iterable[tuple[Any, ...]],
         *,
         dtype: Dtype | None = None,
         copy: bool = False,

--- a/pandas-stubs/core/indexes/base.pyi
+++ b/pandas-stubs/core/indexes/base.pyi
@@ -60,6 +60,7 @@ from pandas.core.indexes.category import CategoricalIndex
 from pandas.core.indexes.datetimes import DatetimeIndex
 from pandas.core.indexes.frozen import FrozenList
 from pandas.core.indexes.interval import IntervalIndex
+from pandas.core.indexes.multi import MultiIndex
 from pandas.core.indexes.period import PeriodIndex
 from pandas.core.indexes.timedeltas import TimedeltaIndex
 from pandas.core.series import Series
@@ -314,6 +315,16 @@ class Index(IndexOpsMixin[S1], ElementOpsMixin[S1]):
         name: Hashable = None,
         tupleize_cols: bool = True,
     ) -> IntervalIndex[Interval]: ...
+    @overload
+    def __new__(
+        cls,
+        data: Sequence[tuple[Any, ...]],
+        *,
+        dtype: Dtype | None = None,
+        copy: bool = False,
+        name: Hashable = None,
+        tupleize_cols: Literal[True] = True,
+    ) -> MultiIndex: ...
     # generic overloads
     @overload
     def __new__(

--- a/tests/indexes/test_indexes.py
+++ b/tests/indexes/test_indexes.py
@@ -1888,7 +1888,7 @@ def test_multiindex_get_loc() -> None:
 
 
 def test_index_from_tuples() -> None:
-    # GH#1498: pd.Index with a sequence of tuples should return MultiIndex
+    """Test creating pd.Index with a sequence of tuples should return MultiIndex."""
     check(
         assert_type(pd.Index([(1, 2), (3, 4)]), pd.MultiIndex),
         pd.MultiIndex,

--- a/tests/indexes/test_indexes.py
+++ b/tests/indexes/test_indexes.py
@@ -1627,7 +1627,7 @@ def test_index_set_names() -> None:
         assert_type(mi.set_names({"elk": "beluga", "owl": "pig"}), pd.MultiIndex),
         pd.MultiIndex,
     )
-    mi = cast("pd.MultiIndex", pd.Index([(1,)]))
+    mi = assert_type(pd.Index([(1,)]), pd.MultiIndex)
     check(assert_type(mi.set_names(1), pd.MultiIndex), pd.MultiIndex, tuple)
 
 
@@ -1885,3 +1885,28 @@ def test_multiindex_get_loc() -> None:
     mi_unsorted = pd.MultiIndex.from_arrays([["a", "b", "a"], ["d", "e", "f"]])
     loc_mask = mi_unsorted.get_loc("a")
     check(assert_type(loc_mask, int | slice | np_1darray_bool), np.ndarray, np.bool_)
+
+
+def test_index_from_tuples() -> None:
+    # GH#1498: pd.Index with a sequence of tuples should return MultiIndex
+    check(
+        assert_type(pd.Index([(1, 2), (3, 4)]), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(pd.Index(UserList([(1, 2), (3, 4)])), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(pd.Index(deque([(1, 2), (3, 4)])), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(pd.Index([("a", 1), ("b", 2)]), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    # tupleize_cols=False: should not be MultiIndex
+    check(
+        assert_type(pd.Index([(1, 2), (3, 4)], tupleize_cols=False), pd.Index),
+        pd.Index,
+    )

--- a/tests/indexes/test_indexes.py
+++ b/tests/indexes/test_indexes.py
@@ -1905,6 +1905,15 @@ def test_index_from_tuples() -> None:
         assert_type(pd.Index([("a", 1), ("b", 2)]), pd.MultiIndex),
         pd.MultiIndex,
     )
+    check(
+        assert_type(pd.Index(frozenset(((1,), (2,)))), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(pd.Index(((1,), (2,))), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+
     # tupleize_cols=False: should not be MultiIndex
     check(
         assert_type(pd.Index([(1, 2), (3, 4)], tupleize_cols=False), pd.Index),

